sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DefaultAvailabilityStrategy.java

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The type Availability strategy.
+ */
+public class DefaultAvailabilityStrategy extends AvailabilityStrategy {
+    /**
+     * The Effective retry regions.
+     */
+    /**
+     * The Number of regions to try.
+     */
+    int numberOfRegionsToTry = 1;
+
+    @Override
+    public List<String> getEffectiveRetryRegions(List<String> preferredRegions, List<String> excludeRegions) {
+        if (excludeRegions == null) {
+            return preferredRegions;
+        }
+        // return preferredRegions without excludeRegions
+        return preferredRegions.stream()
```

Comment:
should the effectiveRetryRegions also exclude the regions which has been marked unavailable?

Reply:
I feel that the logic that picks only available regions out of effective regions should be separate?

Reply:
And this selection from available regions happens in ReplicatedResourceClient